src/game.py

```diff
@@ -14,13 +16,38 @@ def __init__(self, ui=UserInterface(), message=Message()):
         self.rules = Rules()
         self.validator = Validator()
         self.symbol = SymbolOptions()
+        self.computer_player = ComputerPlayer()
         self.ui = ui
         self.set_language(message)
         self.game_board = self.board.starter_board
         self.player_one = config["player_one"]
         self.player_two = config["player_two"]
         self.total_marks_on_board = 0
         self.playing = True
+        self.play_against_computer = False
+
+    def valid_computer_move(self):
+        computer_move = self.computer_player.computer_input()
+        valid_computer_move = self.validator.spot_is_available(
+            self.game_board, computer_move
+        )
+        while not valid_computer_move:
+            computer_move = self.computer_player.computer_input()
+            valid_computer_move = self.validator.spot_is_available(
+                self.game_board, computer_move
+            )
```

Comment:
I know I sound like a broken record but, could you reduce duplication with recursion here?

Reply:
Take a look - I think I did it correctly?!

Reply:
all good!